python/pyarrow/dataset.py

```diff
@@ -741,6 +746,12 @@ def write_dataset(data, base_dir, format=None, partitioning=None, schema=None,
         )
 
     format = _ensure_format(format)
+    if basename_template is None:
+        basename_template = "dat_{i}." + format.default_extname
```

Comment:
Maybe we could go with `"part-{i}."` for the default name (that seems more in line with eg dask / spark) 
(I can also do this in a follow-up PR, if we want this)

Reply:
Not a problem, I'll push this shortly